remote-api/src/main/java/org/alfresco/rest/workflow/api/impl/MapBasedQueryWalker.java

```diff
@@ -247,25 +248,31 @@ else if (type == WhereClauseParser.LESSTHANOREQUALS)
             }
             else
             {
-                throwError = !handleUnmatchedComparison(type, propertyName, propertyValue);
+                throwError = !handleUnmatchedComparison(type, propertyName, propertyValue);
             }
         }
         else
         {
             throwError = !handleUnmatchedComparison(type, propertyName, propertyValue);
         }
 
-        if (throwError) 
-        { 
-            throw new InvalidArgumentException("framework.exception.InvalidProperty", new Object[] {propertyName, propertyValue, WhereClauseParser.tokenNames[type]});
-        }
-        else if (negated)
-        {
-            // Throw error for the unsupported negation only if the property was valid for comparison, show the more meaningful error first.
-            throw new InvalidArgumentException("Cannot use NOT for " + WhereClauseParser.tokenNames[type] + " comparison."); 
+        if (throwError)
+        {
+            throw new InvalidArgumentException("framework.exception.InvalidProperty", new Object[] {propertyName, propertyValue, WhereClauseParser.tokenNames[type]});
+        }
+        else if (negated)
+        {
+            // Throw error for the unsupported negation only if the property was valid for comparison, show the more meaningful error first.
+            throw new InvalidArgumentException("Cannot use NOT for " + WhereClauseParser.tokenNames[type] + " comparison.");
         }
     }
 
+    @Override
+    public Collection<String> getProperty(String propertyName, int type, boolean negated)
+    {
+        return Set.of(this.getProperty(propertyName, type));

```

Comment:
What does `negated` mean in this method? Why is it ignored?

Reply:
This field defines if the way of comparison (let say "EQUALS") should be negated - "NOT EQUALS". Class `MapBasedQueryWalker` doesn't support `NOT` operations at all, and this is the reason why this field is ignored. Maybe an exception should be thrown for `negated == true` 🤔

Reply:
Either that or just make the javadoc clearer that negated will be ignored?

Reply:
Added javadoc with explanation.